internal/backend/auth/authhttpmiddleware/authmiddleware.go

```diff
@@ -27,145 +28,141 @@ var Configs = configset.Set[Config]{
 	Dev:     &Config{UseDefaultUser: true},
 }
 
-// The middleware passes around internally the authenticated user ID,
-// so that we can check only once in the AuthMiddlewareDecorator for
-// the common stuff - that the user exists and that it is not disabled.
-// The AuthMiddlewareDecorator will call eventually the authcontext.SetAuthnUser
-// if the user is deemed worthy of authentication.
-type userIDCtxKey string
-
-var userIDContextKey = userIDCtxKey("authn_user_id")
-
-func ctxWithUserID(ctx context.Context, id sdktypes.UserID) context.Context {
-	return context.WithValue(ctx, userIDContextKey, id)
-}
-
-func getCtxUserID(ctx context.Context) sdktypes.UserID {
-	v := ctx.Value(userIDContextKey)
-	if v == nil {
-		return sdktypes.InvalidUserID
-	}
-
-	return v.(sdktypes.UserID)
-}
-
 type AuthMiddlewareDecorator func(http.Handler) http.Handler
 
 type Deps struct {
 	fx.In
 
+	L        *zap.Logger
 	Cfg      *Config
 	Users    sdkservices.Users
 	Sessions authsessions.Store `optional:"true"`
 	Tokens   authtokens.Tokens  `optional:"true"`
 }
 
-// ifAuthenticated is a middleware that checks if the user is authenticated.
-// If the user is authenticated, it calls the `yes` handler, otherwise it calls the `no` handler.
-func ifAuthenticated(yes, no http.Handler) http.HandlerFunc {
-	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		next := no
+type middlewareError struct {
+	code int
+	msg  string
+}
 
-		if getCtxUserID(r.Context()).IsValid() {
-			next = yes
-		}
+func (m *middlewareError) apply(w http.ResponseWriter) { http.Error(w, m.msg, m.code) }
 
-		next.ServeHTTP(w, r)
-	})
+func newMiddlewareError(code int, msg string) *middlewareError {
+	return &middlewareError{code: code, msg: msg}
 }
 
-func newTokensMiddleware(next http.Handler, tokens authtokens.Tokens) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		authHdr := r.Header.Get("Authorization")
+// middlewareFn is a function that extracts a user ID from a request.
```

Comment:
The name `middleware` is for wrappers around HTTP handlers. This should be called `extractUID` or something like that.

Also, it's better to pass `context.Context` and not the request.

Reply:
> Also, it's better to pass context.Context and not the request.

Whaaaaa? I don't follow here why